docs/clients/promtail/pipelines.md

```diff
@@ -79,63 +87,87 @@ scrape_configs:
+
+The second `match` stage will only run if a label named `app` equals `some-app`, it
+then parses the log line and creates an extracted key named panic if it finds
+`panic: ` in the log line. Then a metrics stage will increment a counter if the
+extracted key `panic` is found in the `extracted` map.
+
+### Data Accessible to Stages
 
-The second `match` stage will only run if a label named `app` == `some-app`, it then parses the log line and creates an extracted key named panic if it finds `panic: ` in the log line.  Then a metrics stage will increment a counter if the extracted key `panic` is found in the `extracted` map.
+The following sections further describe the types that are acessible to each
+stage (although not all may be used)
 
-More info on each field in the interface:
+##### Label Set
 
-##### labels model.LabelSet
+A set of prometheus style labels which will be sent with the log line and will
+be indexed by Loki.
 
-A set of prometheus style labels which will be sent with the log line and will be indexed by Loki.
+##### Extracted Map
 
-##### extracted map[string]interface{}
+Metadata extracted during the pipeline execution which can be used by subsequent
+stages. This data is not sent with the logs and is dropped after the log entry
+is processed through the pipeline.
 
-metadata extracted during the pipeline execution which can be used by subsequent stages.  This data is not sent with the logs and is dropped after the log entry is processed through the pipeline.
+For example, stages like [regex](#regex) and [json](#json) will use expressions
+to extract data from a log line and store it in the `extracted` map, which
+following stages like [timestamp](#timestamp) or [output](#output) can use to
+manipulate the log lines `time` and `entry`.
 
-For example, stages like [regex](#regex) and [json](#json) will use expressions to extract data from a log line and store it in the `extracted` map, which following stages like [timestamp](#timestamp) or [output](#output) can use to manipulate the log lines `time` and `entry`.
+Defined in Go as `map[string]interface{}`.
 
-##### time *time.Time
+##### Log Timesamp
 
-The timestamp which loki will store for the log line, if not set within the pipeline using the [timestamp](#timestamp) stage, it will default to time.Now().
+The timestamp for which Loki will store for the log line, if not set within the
+pipeline using the [timestamp](#timestamp) stage, it will default to time.Now().
 
-##### entry *string
+##### Log Line
 
-The log line which will be stored by loki, the [output](#output) stage is capable of modifying this value, if no stage modifies this value the log line stored will match what was input to the system and not be modified.
+The log line which will be stored by loki, the [output](#output) stage is
+capable of modifying this value, if no stage modifies this value the log line
+stored will match what was input to the system and not be modified.
 
 ## Stages
```

Comment:
May be worth splitting each stage documentation in its own dedicated file, in order to be able to show extensive examples without the risk of overloading this file?